homeassistant/components/mqtt/models.py

```diff
@@ -289,6 +295,30 @@ def async_render_with_possible_json_value(
         return rendered_payload
 
 
+class EntityMonitor:
+    """Monitors entity state changes."""
+
+    _singleton: UndefinedType = UNDEFINED
+
+    def __init__(self, entity: Entity, attributes: set[str]) -> None:
+        """Initialize entity monitor."""
+        self._entity: Entity = entity
+        self._attributes: dict[str, Any] = {
+            attribute: getattr(entity, attribute, self._singleton)
+            for attribute in attributes
+        }
+
+    @property
+    def changed(self) -> bool:
+        """Return True if attributes on entity changed or if update is forced."""
+        if getattr(self._entity, "_attr_force_update", False):
```

Comment:
```suggestion
        entity = self._entity
        if getattr(entity, "_attr_force_update", False):
```

Not sure if it will make a difference, but you can save all the `self` `.` `_entity` lookups here